pydax/_schema.py

```diff
@@ -109,24 +119,47 @@ def add_schema(self, name: str, val: Schema) -> None:
         self.schemata[name] = val
 
 
-def load_schemata(*,
-                  dataset_url: Optional[str] = None,
-                  format_url: Optional[str] = None,
-                  license_url: Optional[str] = None) -> SchemaManager:
-    """Helper function to load and provide all schemata.
+# The SchemaManager object that is managed by high-level functions
+_schemata: Optional[SchemaManager] = None
+
+
+def load_schemata(*, force_reload: bool = False) -> None:
+    """Loads a :class:`SchemaManager` object that stores all schemata. To export the loaded :class:`SchemaManager`
+    object, please use :func:`.export_schemata`.
 
-    :param dataset_url: Dataset schema url, defaults to DatasetSchema.DEFAULT_SCHEMA_DATASETS_URL
-    :param format_url: Format schema url, defaults to FormatSchema.DEFAULT_SCHEMA_FORMATS_URL
-    :param license_url: License schema url, defaults to LicenseSchema.DEFAULT_SCHEMA_LICENSES_URL
-    :return: A :class:`SchemaManager` object which holds the loaded schemata in :attr:`schemata`
+    :param force_reload: If ``True``, force reloading even if the provided URLs by :func:`pydax.init` are the same as
+         provided last time. Otherwise, only those that are different from previous given ones are reloaded.
     """
-    if dataset_url is None:
-        dataset_url = get_config().DEFAULT_DATASET_SCHEMA_URL
-    if format_url is None:
-        format_url = get_config().DEFAULT_FORMAT_SCHEMA_URL
-    if license_url is None:
-        license_url = get_config().DEFAULT_LICENSE_SCHEMA_URL
-
-    return SchemaManager(dataset_schema=DatasetSchema(dataset_url),
-                         format_schema=FormatSchema(format_url),
-                         license_schema=LicenseSchema(license_url))
+    urls = {
+        'datasets': get_config().DATASET_SCHEMA_URL,
+        'formats': get_config().FORMAT_SCHEMA_URL,
+        'licenses': get_config().LICENSE_SCHEMA_URL
+    }
+
+    global _schemata
+    if force_reload or _schemata is None:  # Force reload or clean slate, create a new SchemaManager object
+        _schemata = SchemaManager(**{name: Schema(url) for name, url in urls.items()})
+    else:
+        for name, schema in _schemata.schemata.items():
+            if schema.retrieved_url_or_path != urls[name]:
+                _schemata.add_schema(name, Schema(urls[name]))
+
+
+def get_schemata() -> SchemaManager:
```

Comment:
```suggestion
def _get_schemata() -> SchemaManager:
```
Should this be a private function then since its only to be internally used?

Reply:
Do we even need `get_schemata`? Seems like we can combine `get_schemata` inside of `export_schemata`? I guess I can see `get_schemata` being useful for tests.

Reply:
We expose `export_schemata` but not `get_schemata`, so I don't think we can really combine them into one function. I added `get_schemata` mainly for situations we need to alter the schemata file and performance concerns when used internally (this may not matter much).

This module starts with an underscore, and this function is not exposed, so it is already limited for internal use. When the module starts with an underscore and a function in it doesn't, it signals that the function is not used solely by the module itself and other modules are welcome to use this function.

Reply:
I updated the comment of this function to clarify